Highway.h

```diff
@@ -2,20 +2,26 @@
 
 #include <vector>
 #include <algorithm>
+
+#include "Car.h"
+#include "Motorcycle.h"
+#include "SemiTruck.h"
```

Comment:
You aren't using these here.  You're using them in `Highway.cpp`.   
- [ ] move these `#include` statements and forward declare instead.